### ISSUE TITLE
Best Prompt 

### ISSUE THREAD

**Issue by egeozsoy** [yangyangyang127/PointCLIP_V2#4]
How did you decide the best prompt (https://github.com/yangyangyang127/PointCLIP_V2/blob/main/zeroshot_cls/trainers/best_param.py) from the many prompts listed here (https://github.com/yangyangyang127/PointCLIP_V2/blob/main/zeroshot_cls/prompts/modelnet40_1000.json)?

I went through the code and the paper, but could not find a mention of this

**Comment by zou-longkun** [comment]
Hi，I have similar doubts. If you choose the best prompts on the test set or verification set, isn't it equivalent to using labels indirectly? Strictly speaking, zero-shot settings should not use any sample labels.

**Comment by egeozsoy** [comment]
> Hi，I have similar doubts. If you choose the best prompts on the test set or validation set, isn't it equivalent to using labels indirectly? Strictly speaking, zero-shot settings should not use any sample labels.

strictly speaking, no setting should use „test“ labels

**Comment by ZrrSkywalker** [comment]
Thanks for your valuable suggestion. @egeozsoy @zou-longkun 

1. We follow the `zero-shot setting of all open-world CLIP-based methods`, e.g., CoOp, CLIP-Adapter in 2D, and CLIP2Point, ULIP in 3D. They need the validation set to tune the best hyperparameters, such as CoOp's learnable prompt length, CLIP-Adapter's architectures, or CLIP2Point's 3D prompts. 
2. Like the popular ImageNet dataset, 3D datasets (ModelNet40, ScanObjectNN) share the validation and test set, so we tune our prompt by the test/val set just as previous ImageNet methods (ViT, MAE) or 3D methods (PointNet, PointMLP).
3. More specifically, `CLIP itself` tunes a different prompt on different downstream tasks for `zero-shot classification`. For example, on ImageNet, CLIP utilizes an ensemble of seven handcraft templates for the best test-set performance.

As aforementioned, every method requires a validation set for tuning to best achieve its potential, even the CLIP-based zero-shot setting that we followed. If we split a sub-validation set for prompt tuning from the training set of ModelNet40 or ScanObjectNN, the zero-shot result of PointCLIP V2 is still competitive. We will show such experiment in a few days.

Thanks again for your insightful comment that makes our work better!
